fenix/app/src/main/java/org/mozilla/fenix/components/appstate/AppAction.kt

```diff
@@ -30,6 +31,28 @@ import org.mozilla.fenix.wallpapers.Wallpaper
  * [Action] implementation related to [AppStore].
  */
 sealed class AppAction : Action {
+    /**
+     * [AppAction] dispatched to indicate that the store is initialized and
+     * ready to use. This action is dispatched automatically before any other
+     * action is processed. Its main purpose is to trigger initialization logic
+     * in middlewares. The action itself should have no effect on the [AppState].
+     */
+    object Init : AppAction()
+
+    /**
+     * NOTE: This action is not yet functional and will require https://bugzilla.mozilla.org/show_bug.cgi?id=1845409
+     * to be resolved. This is part of an ongoing lib-state refactor.
```

Comment:
I'm open to other suggestions here, but this felt like the appropriate name for what was occurring at callsites. They tend to dispatch this action, and then call `HomeActivity::openToBrowserAndLoad`. Ideally, that function will be handled by middleware that react to this action in an upcoming patch.